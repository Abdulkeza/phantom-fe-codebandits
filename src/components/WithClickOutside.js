import React, { useCallback, useEffect, useRef, useState } from 'react';
import { useLocation } from 'react-router-dom';

const WithClickOutside = (WrappedComponent) => {
  const Component = () => {
    let location = useLocation();
    const [open, setOpen] = useState(false);
    const ref = useRef(null);
    const handleClickOutside = (e) => {
      !ref.current.contains(e.target) && setOpen(false);
    };
    const setRef = useCallback((node) => {
      ref.current && document.addEventListener('mousedown', handleClickOutside);

      return [ref];
    }, []);

    useEffect(() => {
      open && setOpen(false);
    }, [location.key]);
    const [newRef] = setRef();

    return <WrappedComponent open={open} setOpen={setOpen} ref={newRef} />;
  };

  return Component;
};

export default WithClickOutside;
