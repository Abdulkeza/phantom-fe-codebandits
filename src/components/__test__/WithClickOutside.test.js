import React from 'react';
import renderer from 'react-test-renderer';
import WithClickOutside from '../WithClickOutside.js';

const wrapper = renderer.create(<WithClickOutside />);

describe('withClickOutside', () => {
  it('should render withClickOutside', () => {
    const elem = renderer.create(<WithClickOutside />).toJSON();
    expect(elem).toMatchSnapshot();
  });
});

describe('<WithClickOutside /> functions', () => {
  it('test the Component function', () => {
    const Component = jest.fn(wrapper.getInstance(), 'Component');
    expect(Component).toMatchSnapshot();
  });
});
