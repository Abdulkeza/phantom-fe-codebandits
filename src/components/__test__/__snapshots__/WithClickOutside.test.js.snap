// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WithClickOutside /> functions test the Component function 1`] = `[MockFunction]`;

exports[`withClickOutside should render withClickOutside 1`] = `null`;
